### PR TITLE
Adding nanoprecision to protobuf timestamp response

### PR DESCRIPTION
**Pull request by majtsaxo**
Relates to any timestamp given that should be comparable by nano precision.

### CODE REVIEW

File: src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js [hunk 1]
```diff
@@ -7,10 +7,34 @@ export default {
...
+                    const NANOS_PER_MILLISECOND = 1000000;
+                    const NANOS_PER_MICROSECOND = 1000;
 
-                    return date.toJSON();
+                    const date = new Date(Number(seconds) * MILIS_PER_SECOND);
+
+                    let result = date.toJSON();
+
+                    // Support for Nano precision
+                    if (nanos !== 0) {
```
Comment (michax) [inline src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js]: Nice find. Thanks!

Could you add some test cases to serializer-protobuf-spec.js, with mocked nanoseconds in proto-price.js?

File: src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js [hunk 2]
```diff
@@ -7,10 +7,34 @@ export default {
                 },
 
                 toObject(message, options) {
-                    const seconds = Number(message.seconds) * 1000;
-                    const date = new Date(seconds);
+                    const { seconds, nanos } = message;
+                    const MILIS_PER_SECOND = 1000;
```
Comment (michax) [inline src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js]: Maybe we can simplify it. Check google timestamp wrappere for JS:
https://github.com/google/protobuf/blob/master/src/google/protobuf/compiler/js/well_known_types/timestamp.js

File: src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js (hunk 1 above)
Comment (majtsaxo) [reply src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js]: Yes, will do.

File: src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js (hunk 2 above)
Comment (majtsaxo) [reply src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js]: Good idea!

File: src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js [hunk 3]
```diff
@@ -7,8 +7,10 @@ export default {
                 },
 
                 toObject(message, options) {
-                    const seconds = Number(message.seconds) * 1000;
-                    const date = new Date(seconds);
+                    const { seconds, nanos } = message;
+
+                    // Date with support for nano precision
+                    const date = new Date((Number(seconds) * 1000) + (Number(nanos) / 1000000));
```
Comment (lukeapage) [inline src/openapi/streaming/serializer/wrappers/protobuf-wrappers.js]: Shall we adjust this to be an integer? e.g.

Math.floor(Number(nanos) / 1000000)

so its clear this is to get the ms, not that Date supports fractions..

**Pull request merged**